lib/ansible/modules/network/fortios/fortios_api.py

```diff
@@ -0,0 +1,164 @@
+---
+module: fortios_api
+extends_documentation_fragment: fortios_api_doc
+version_added: "2.5"
+short_description: Manages arbitrarily designated fortios configurations.
+description:
+    - Manages arbitrarily designated fortios configurations. Can be used to edit any endpoint, as long as proper API path is provided.
+author:
+    - Will Wagner (@willwagner602)
+      Eugene Opredelennov (@eoprede)
+notes:
+    - Tested against Fortigate v5.4.5 VM
+
+options:
+    endpoint_information:
+      description: Information about the endpoint.
+      required: true
+      suboptions:
+            endpoint:
+                required: true
+                description: identifies the string to be added to the api
+                             URL to access the appropriate endpoint.
+            list_identifier:
+                required: true
+                description: The identifier used by this configuration to identify the list of
+                             objects that should be used to update the endpoint's configuration.
+            object_identifier:
+                required: false
+                description: The unique identifier used by this endpoint to differentiate objects.
+                             Typically 'name', 'id', or some variation. Not required for endpoints which
+                             contain only a single object.
+            permanent_objects:
+                required: false
+                description: A list of identifiers for objects at the endpoint that cannot be deleted.
+
+'''
+EXAMPLES = '''
+- name: Update router static
+  fortios_api:
+    endpoint_information:
+      endpoint: cmdb/router/static
+      list_identifier: routers
+      object_identifier: seq-num
+      permanent_objects:
+    conn_params:
+      fortigate_username: admin
+      fortigate_password: test
+      fortigate_ip: 1.2.3.4
+      port: 10080
+      verify: false
+      secure: false
+      proxies:
+          http: socks5://127.0.0.1:9000
+    routers:
+    - seq-num: 1
+      status: enable
+      dst: 0.0.0.0 0.0.0.0
+      gateway: 192.0.2.1
+      distance: 10
+      weight: 0
+      priority: 0
+      device: port1
+      comment: ''
+      blackhole: disable
+      dynamic-gateway: disable
+      virtual-wan-link: disable
+      dstaddr: ''
+      internet-service: 0
+      internet-service-custom: ''
+
+    - seq-num: 2
+      status: enable
+      dst: 0.0.0.0 0.0.0.0
+      gateway: 192.0.3.1
+      distance: 10
+      weight: 0
+      priority: 0
+      device: port2
+      comment: ''
+      blackhole: disable
+      dynamic-gateway: disable
+      virtual-wan-link: disable
+      dstaddr: ''
+      internet-service: 0
+      internet-service-custom: ''
+'''
+
+RETURN = '''
+proposed:
+    description: k/v pairs of parameters passed into module and sent to the device for changes
+    returned: always
+    type: list
+    sample: '[{"blackhole": "disable", "comment": "", "device": "port1", "distance": 10,
```

Comment:
In `RETURN` blocks we now support nested returns (eg returning complex data structures, see http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#return-block

Reply:
Thanks for the info. I am not entirely sure if there's any benefit to using complex data structures in documentation. Also, the returns are different per different API endpoint, so samples will vary greatly (and I think in some other modules we don't even include them, as the response may be a few pages long)